pkg/provisioner/ironic/ironic.go

```diff
@@ -1011,8 +1020,21 @@ func (p *ironicProvisioner) Provision(getUserData provisioner.UserDataSource) (r
 		return p.startProvisioning(ironicNode, checksum, getUserData)
 
 	case nodes.Manageable:
+		if len(cleanSteps) > 0 && len(ironicNode.CleanStep) == 0 {
```

Comment:
This logic is only being applied in one place. What happens if the deployment fails? Shouldn't we retry the cleaning?

I think if you move this logic into startProvisioning() it will be more clear that wiping any old RAID configuration and preparing a new one is part of the provisioning process. It will also address my concern about whether the cleaning work is being done at all of the right places.

Reply:
+1
Thanks. I've move `manualCleaning()` into `startProvisioning()` to apply in both `manageable` and `deployfaild` state